src/System.Collections/src/System/Collections/StructuralComparisons.cs

```diff
@@ -70,7 +70,7 @@ public int GetHashCode(object obj)
         {
             if (obj == null) return 0;
 
-            IStructuralEquatable seObj = obj as IStructuralEquatable;
+            IStructuralEquatable? seObj = obj as IStructuralEquatable;
```

Comment:
should obj be nullable?

Reply:
I'm going to leave it as non-nullable.  We say that IEqualityComparer.GetHashCode should accept a non-null object, and the docs agree.  So this is unnecessary handling null.  It's also not public, returned and only exposed via the interface.